### DISCUSSION TITLE
How to calculate similarity between the two graphs?

### DISCUSSION THREAD

**Discussion by smith-co** [pyg-team/pytorch_geometric#4914] [Q&A]
I have to learn similarity between graphs using deep learning. I have many samples (~500k) of graphs. Graphs have ~5000 nodes and ~4000 edges in the average.

How can I compute similarity score between two graphs? I am thinking:

- convert graphs into vectors using Graph2Vec embedding
- then compare them using various similarity calculating techniques like cosine similarity.

I would really appreciate if I can get some feedback whether this is the correct way to approach this problem or not.

**Comment by rusty1s** [comment]
If you have some signal about positive and negative samples, you could use a contrastive loss/triplet loss to learn this task. Otherwise, you can learn in an unsupervised fashion, e.g., via Deep Graph Infomax or other methods. Your general way to approach this problem is definitely correct.

**Comment by UlviShukurzade** [comment]
hey, how was your outcome, what have ended up using?
